### PR TITLE
Allow multiple builders by the same name, per aspect kind.

### PR DESCRIPTION
**Pull request by geeknoid**
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/227)

### CODE REVIEW

File: pkg/adapterManager/registry.go [hunk 1]
```diff
@@ -20,20 +20,19 @@ import (
 	"github.com/golang/glog"
 	"istio.io/mixer/pkg/adapter"
 	"istio.io/mixer/pkg/aspect"
-	"istio.io/mixer/pkg/config"
 )
 
 // registry implements pkg/adapter/Registrar.
 // registry is initialized in the constructor and is immutable thereafter.
 // All registered builders must have unique names.
```
Comment (ZackButcher) [inline pkg/adapterManager/registry.go]: Update this comment -> "All registered builders must have a unique name per aspect kind"

File: pkg/adapterManager/registry.go [hunk 2]
```diff
@@ -20,20 +20,19 @@ import (
...
 // It also implements builderFinder that manager uses.
 type registry struct {
-	buildersByName map[string]adapter.Builder
+	builders map[string]map[string]adapter.Builder
```
Comment (ZackButcher) [inline pkg/adapterManager/registry.go]: Can you add a comment noting that the outer map is by kind and the inner map is by name? The types aren't helpful here.

Comment (douglas-reid) [inline pkg/adapterManager/registry.go]: Maybe we could have a type named KindNameBuilderMap or something to document it here and the other places in the code?

File: pkg/aspect/inventory.go [hunk 3]
```diff
@@ -18,18 +18,22 @@ import (
...
-
 // APIBinding associates an aspect with an API method
 type APIBinding struct {
 	Aspect Manager
 	Method config.APIMethod
 }
 
+// Name of all supported aspect kinds.
+const (
+	AccessLogKind = "access-logs"
```
Comment (douglas-reid) [inline pkg/aspect/inventory.go]: i feel like we need/want the tautological comments for these exported consts here.

File: pkg/adapterManager/registry.go (hunk 1 above)
Comment (geeknoid) [inline pkg/adapterManager/registry.go]: Done

File: pkg/adapterManager/registry.go (hunk 2 above)
Comment (geeknoid) [inline pkg/adapterManager/registry.go]: Done

File: pkg/aspect/inventory.go (hunk 3 above)
Comment (geeknoid) [inline pkg/aspect/inventory.go]: What do you mean?

Comment (douglas-reid) [inline pkg/aspect/inventory.go]: I was guessing that it was a lint issue (not documenting an exported element). I _feel_ like that used to be a golint issue, but maybe not now? or maybe my memory is hazy?

**Comment by geeknoid** [comment]
I didn't have a comment at first, and one of the linters said I needed a
comment either on each element or on the block as a whole. Putting the
comment on the block as a whole keeps all the linters happy.

File: testdata/globalconfig.yml [hunk 4]
```diff
@@ -2,16 +2,16 @@ subject: "namespace:ns"
 revision: "2022"
 adapters:
   - name: default
-    kind: istio/quota
-    impl: istio/memQuota
+    kind: quotas
```
Comment (mandarjog) [inline testdata/globalconfig.yml]: With an adapter providing multiple aspects, kind field should be removed from the proto.

**Pull request merged**